sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -346,6 +346,49 @@ object RapidsPluginUtils extends Logging {
       loadExtensions(classOf[SparkPlugin], pluginClasses)
     }
   }
+
+  /**
+   * Extracts supported GPU architectures from the given properties file
+   */
+  private def getSupportedGpuArchitectures(propFileName: String): Set[Int] = {
+    val props = RapidsPluginUtils.loadProps(propFileName)
+    Option(props.getProperty("gpu_architectures"))
+      .getOrElse(throw new RuntimeException(s"GPU architectures not found in $propFileName"))
+      .split(";")
+      .map(_.toInt)
+      .toSet
+  }
+
+  /**
+   * Checks if the current GPU architecture is supported by the spark-rapids-jni
+   * and cuDF libraries.
+   *
+   * Compatibility Rule: Binaries generated for a certain compute capability is supported
+   * to run on any GPU with the same major revision and same or higher minor revision of
+   * compute capability.
+   *
+   * See: https://docs.nvidia.com/cuda/ampere-compatibility-guide/index.html
+   */
+  def checkGpuArchitecture(): Unit = {
+    val supportedGpuArchsSet = getSupportedGpuArchitectures(JNI_PROPS_FILENAME)
+      .intersect(getSupportedGpuArchitectures(CUDF_PROPS_FILENAME))
+    val supportedArchsStr = supportedGpuArchsSet.toSeq.sorted.mkString(", ")
+    val majorGpuArch = Cuda.getComputeCapabilityMajor
+    val minorGpuArch = Cuda.getComputeCapabilityMinor
+    // Check if the device's major architecture is supported.
+    val supportedMajorArchs = supportedGpuArchsSet.filter(ver => majorGpuArch == ver / 10)
+    if (supportedMajorArchs.isEmpty) {
+      throw new RuntimeException(s"Device architecture $majorGpuArch$minorGpuArch is unsupported." +
+        s" Supported architectures: $supportedArchsStr.")
+    }
+    // Check if the device's minor architecture is supported. See docs for explanation.
+    val supportedMinorArchs = supportedMajorArchs.filter(ver => minorGpuArch >= ver % 10)
+    if (supportedMinorArchs.isEmpty) {
+      logWarning(s"Device architecture $majorGpuArch$minorGpuArch is supported based on the" +
+        s" major version. Supported architectures: $supportedArchsStr. " +
+        s"See https://docs.nvidia.com/cuda/ampere-compatibility-guide/index.html")
+    }
```

Comment:
No need to do a full filter here and a second pass, as we can early out in one pass.  Also the original code does not fail in cases it needs to (e.g.: GPU arch is 70 but lowest supported is 75).
```suggestion
    val isSupported = supportedGpuArchSet.exists { ver =>
      majorGpuArch == ver / 10 && minorGpuArch >= ver % 10
    }
    if (!isSupported) {
      throw new RuntimeException(s"Device architecture $majorGpuArch$minorGpuArch is unsupported." +
        s" Supported architectures: $supportedArchsStr.")
    }
```

Reply:
Added unit tests